### DISCUSSION TITLE
Buoyancy Support

### DISCUSSION THREAD

**Discussion by JackJ30** [godot-jolt/godot-jolt#512] [Questions]
Does godot-jolt support buoyancy? I haven't found any reference to it on the readme. Anyone know more about this?
Thank you

**Comment by mihe** [comment]
No, there's no support for buoyancy right now.

I haven't really looked into what would be required to support this, but from a quick glance it seems like it's just about exposing Jolt's `Body::ApplyBuoyancyImpulse` method somehow. I'm not entirely sure where you would expose this though, since there's no way to add this to the existing `RigidBody3D` node, and no way to add "static classes" to serve as helper functions either, as far as I'm aware.

It might be more appropriate to introduce a new type of node, like a `JoltFluidVolume3D` or something, that does the necessary query every tick to see which bodies it overlaps and applies the buoyancy accordingly.

I'll think about it for a bit, but I can't say if/when it would be implemented.

**Reply by tjpotts** [reply]
I have a commit at tjpotts/godot-jolt@60e2720bad8f774755c6978f219c1eae6e3f4bb3 which implements a `JoltBuoyancyRigidBody3D`. I'd be happy to submit a PR if this is something you think is worth merging.
